service/src/main/java/bio/terra/workspace/service/resource/controlled/flight/clone/azure/database/DumpAzureDatabaseStep.java

```diff
@@ -0,0 +1,156 @@
+                (wsmResource) ->
+                    wsmResource
+                        .castToControlledResource()
+                        .getAccessScope()
+                        .equals(AccessScopeType.ACCESS_SCOPE_SHARED))
+            .findFirst()
+            .orElseThrow(
+                () ->
+                    new IllegalStateException(
+                        "No storage container to own dumpfile destination workspace %s"
+                            .formatted(destinationWorkspaceId)))
+            .castByEnum(WsmResourceType.CONTROLLED_AZURE_STORAGE_CONTAINER);
+
+    workingMap.put(
+        WorkspaceFlightMapKeys.ControlledResourceKeys.AZURE_STORAGE_CONTAINER,
+        destinationContainer);
+
+    var sasToken =
+        azureStorageAccessService.createAzureStorageContainerSasToken(
+            destinationWorkspaceId, destinationContainer, userRequest, null, null, "rcw");
+    var blobContainerUrlAuthenticated = sasToken.sasUrl();
+
+    var blobFileName =
+        String.format(
+            "%s_%s.dump",
+            sourceDatabase.getDatabaseName(),
+            new SimpleDateFormat("yyyy-MM-dd_HH:mm:ss").format(new java.util.Date()));
+    workingMap.put(WorkspaceFlightMapKeys.ControlledResourceKeys.CLONE_DB_DUMPFILE, blobFileName);
+
+    // Query LZ for the postgres server info
+    var bearerToken = new BearerToken(samService.getWsmServiceAccountToken());
+    var landingZoneId =
+        landingZoneApiDispatch.getLandingZoneId(
+            bearerToken, workspaceService.getWorkspace(sourceDatabase.getWorkspaceId()));
+    var dbServerName =
+        getResourceName(
+            landingZoneApiDispatch
+                .getSharedDatabase(bearerToken, landingZoneId)
+                .orElseThrow(() -> new RuntimeException("No shared database found")));
+    var adminDbUserName =
+        getResourceName(
+            landingZoneApiDispatch
+                .getSharedDatabaseAdminIdentity(bearerToken, landingZoneId)
+                .orElseThrow(
+                    () -> new RuntimeException("No shared database admin identity found")));
+
+    logger.info(
+        "running DumpAzureDatabaseStep with blobContainerName {} and blobFileName {}",
+        destinationContainer.getStorageContainerName(),
+        blobFileName);
+
+    this.azureDatabaseUtilsRunner.pgDumpDatabase(
+        workingMap.get(AZURE_CLOUD_CONTEXT, AzureCloudContext.class),
+        sourceDatabase.getWorkspaceId(),
+        "dump-db-" + this.sourceDatabase.getResourceId(),
+        sourceDatabase.getDatabaseName(),
+        dbServerName,
+        adminDbUserName,
+        blobFileName,
+        destinationContainer.getStorageContainerName(),
+        blobContainerUrlAuthenticated);
+
+    return StepResult.getStepResultSuccess();
+  }
+
+  @Override
+  public StepResult undoStep(FlightContext context) throws InterruptedException {
+    // Nothing to undo in this step (dump file will be cleaned up by storage container deletion)
```

Comment:
is this still true?

Reply:
Not currently, but it will be (hence why we are merging into feature branch).